spyder/widgets/editor.py

```diff
@@ -766,9 +766,9 @@ def open_fileswitcher_dlg(self):
             self.fileswitcher_dlg.hide()
             self.fileswitcher_dlg.is_visible = False
             return
-        self.fileswitcher_dlg = FileSwitcher(self, self.tabs, self.data)
+        self.fileswitcher_dlg = FileSwitcher(self, self.tabs, self.data, self)
         self.fileswitcher_dlg.sig_goto_file.connect(self.set_stack_index)
-        self.fileswitcher_dlg.sig_close_file.connect(self.close_file)
+        self.fileswitcher_dlg.setup()
         self.fileswitcher_dlg.show()
         self.fileswitcher_dlg.is_visible = True

```

Comment:
Do we still need this instance of the file switcher here? I mean, can we remove it given that we have now a single and global instance?